Sources/OpenAPIRuntime/Conversion/MIMEType.swift

```diff
@@ -0,0 +1,156 @@
+        }
+        guard lhs.subtype == rhs.subtype else {
+            return false
+        }
+        // Parameter names are case-insensitive, parameter values are
+        // case-sensitive.
+        guard lhs.parameters.count == rhs.parameters.count else {
+            return false
+        }
+        func normalizeKeyValue(key: String, value: String) -> (String, String) {
+            (key.lowercased(), value)
+        }
+        let normalizedLeftParams = Dictionary(
+            uniqueKeysWithValues: lhs.parameters.map(normalizeKeyValue)
+        )
+        let normalizedRightParams = Dictionary(
+            uniqueKeysWithValues: rhs.parameters.map(normalizeKeyValue)
+        )
+        return normalizedLeftParams == normalizedRightParams
+    }
+}
+
+extension MIMEType.Token: LosslessStringConvertible {
+    public init?(_ description: String) {
+        if description == "*" {
+            self = .wildcard
+        } else {
+            self = .concrete(description)
+        }
+    }
+
+    public var description: String {
+        switch self {
+        case .wildcard:
+            return "*"
+        case .concrete(let string):
+            return string
+        }
+    }
+}
+
+extension MIMEType: LosslessStringConvertible {
+    public init?(_ description: String) {
+        var components =
+            description
+            // Split by semicolon
+            .split(separator: ";")
+            .map(String.init)
+            // Trim leading/trailing spaces
+            .map { $0.soar_trimmingLeadingAndTrailingSpaces }
+        guard !components.isEmpty else {
+            return nil
+        }
+        let firstComponent = components.removeFirst()
+        let typeAndSubtype =
+            firstComponent
+            .split(separator: "/")
+            .map(String.init)
+            .compactMap(MIMEType.Token.init)
+        guard typeAndSubtype.count == 2 else {
+            return nil
+        }
+        func parseParameter(_ string: String) -> (String, String)? {
+            let components =
+                string
+                .split(separator: "=")
+                .map(String.init)
+            guard components.count == 2 else {
+                return nil
+            }
+            return (components[0], components[1])
+        }
+        let parameters =
+            components
+            .compactMap(parseParameter)
+        self.init(
+            type: typeAndSubtype[0],
+            subtype: typeAndSubtype[1],
+            parameters: Dictionary(
+                parameters,
+                uniquingKeysWith: { a, _ in a }
+            )
+        )
+    }
+
+    public var description: String {
+        (["\(type.description)/\(subtype.description)"]
+            + parameters
+            .sorted(by: { a, b in a.key < b.key })
+            .map { "\($0)=\($1)" })
+            .joined(separator: "; ")
+    }
+}
+
+extension String {
+    fileprivate var soar_trimmingLeadingAndTrailingSpaces: Self {
```

Comment:
Oh, I did it because I thought that if our dependencies introduced `trimmingLeadingAndTrailingSpaces`, that it'd break us, but now it's coming back to me that the local module's implementation always wins and doesn't produce a clash error, right?

Reply:
Fixed: https://github.com/apple/swift-openapi-runtime/pull/29/commits/5752e18b1ae469197ae5e38f7533edafb8ddbc10